app/models/course/registration.rb

```diff
@@ -0,0 +1,29 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: roles.values_at(:teaching_assistant, :manager, :owner)) }
```

Comment:
Then a way to do it is to add the wanted key, array pair to the roles enum. I can write a module and add a scoped_enum that calls enum and does the addition then let the class extend the module. Is that fine?